video/out/gpu/video.c

```diff
@@ -4360,11 +4369,49 @@ void gl_video_load_hwdecs(struct gl_video *p, struct mp_hwdec_devices *devs,
     p->hwdec_interop_loading_done = true;
 }
 
-void gl_video_load_hwdecs_all(struct gl_video *p, struct mp_hwdec_devices *devs)
+void gl_video_load_hwdecs_for_img_fmt(struct gl_video *p,
+                                      struct mp_hwdec_devices *devs,
+                                      int imgfmt)
 {
-    if (!p->hwdec_interop_loading_done) {
-        for (int n = 0; ra_hwdec_drivers[n]; n++)
-            load_add_hwdec(p, devs, ra_hwdec_drivers[n], true);
-        p->hwdec_interop_loading_done = true;
+    if (p->hwdec_interop_loading_done) {
```

Comment:
I am not sure if this is a real case we have to think about, but what if the function gets called, but with a different imgfmt (from specific to none or from one specific to another would be the two cases that pop to my mind) ?

If switching of the imgfmt may only happen in case the whole gl_video gets reset, then just having a single boolean is fine. If we could have more and more things getting loaded with various calls, then just boolean would not be sufficient and we'd either have to have a bitmask or an array of bools...

Reply:
I respected this boolean because if it's set, we know we loaded all interops. But if it's not set, we have a separate check for each individual interop to only load it if it's not already been loaded. So we can move between interops safely and also work correctly after a load_all happens.